File: SeleniumDemo/src/noframe/cases/SimpleExampleModified_2.java
package noframe.cases;

import java.util.List;

import org.junit.After;
import org.junit.AfterClass;
import org.junit.Assert;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedCondition;
import org.openqa.selenium.support.ui.WebDriverWait;

import com.thoughtworks.selenium.SeleneseTestBase;

public class SimpleExampleModified_2 {
	private WebDriver driver = null;
	private SeleneseTestBase testbase = null; 
	
	@BeforeClass
	public static void setUpBeforeClass(){
		System.out.println("before class.");
	}
	
	@Before
	public void setUpBeforeMethod() {
		driver=  new FirefoxDriver();
		testbase = new SeleneseTestBase();
		System.out.println("before @test method.");
	}
	
    @Test
	public void assert1212OnSales(){
   	    
        driver.get("http://www.360buy.com/");
        
        (new WebDriverWait(driver, 20)).until(new ExpectedCondition<Boolean>() {
            public Boolean apply(WebDriver d) {
                return d.findElement(By.id("subscription-submit")).isDisplayed();
            }
        });
       
        //by id
        WebElement input_search = driver.findElement(By.id("key")); 
        
        // by xpath
        WebElement button_search = driver.findElement(By.xpath("//*[@id='search']//input[@type='button']"));
        
        //by class name
        WebElement branch_address_1 = driver.findElement(By.className("ar"));
//        System.out.println("by class name: "+branch_address_1.getText());
        
        // by css selector
        WebElement branch_address_2 = driver.findElement(By.cssSelector("p.ar"));
//        System.out.println("by css selector: "+branch_address_2.getText());
        
        // by link text
        WebElement about_us = driver.findElement(By.linkText("关于我们"));
//        System.out.println("by link text: "+about_us.getText());
        
        //by tag name
        WebElement jd_report = driver.findElement(By.id("jdnews")).findElement(By.tagName("h2"));
//        System.out.println("by tag name: "+jd_report.getText());

        input_search.sendKeys("12.12");

        button_search.click();

        (new WebDriverWait(driver, 20)).until(new ExpectedCondition<Boolean>() {
            public Boolean apply(WebDriver d) {
                return d.findElement(By.id("plist")).isDisplayed();
            }
        });

        Assert.assertTrue(driver.getTitle().startsWith("12.12"));
        
        System.out.println("the 1st case");
	}

	@Test
	public void assert1212OnSales_2() {
		System.out.println("the 2nd case");
	}

	@After
	public void tearDownAfterMethod() {
		driver.quit();
		System.out.println("after @test method");
	}

	@AfterClass
	public static void tearDownAfterClass() {
		System.out.println("after class");
	}
    
}

File: PythonDemo/src/demo/SimpleExampleModified_2.py
# -*- coding:UTF-8 -*-

'''
Created on 2012-12-18

@author: chengkai
'''

from selenium import webdriver
from selenium.webdriver.support.ui import WebDriverWait 
from selenium.common.exceptions import ElementNotVisibleException
import time
import unittest

class SearchPoductsTest(unittest.TestCase):
    
    def setUp(self):
#        driver= webdriver.Chrome()
        self.driver= webdriver.Firefox()
        print 'run before test.'
   
    def tearDown(self):
        self.driver.quit()
        print 'run after test.'

    def testProductsNameEqual(self):
        
        self.driver.get('http://www.360buy.com/')
        WebDriverWait(self.driver, 20, 1, (ElementNotVisibleException)).until
        (lambda driber: self.driver.find_element_by_id('subscription-submit').is_displayed())
        initiate = self.initiateElement
        initiate()
        self.input_search.send_keys("12.12")
        self.button_search.click()
       
        WebDriverWait(self.driver, 20, 1, (ElementNotVisibleException)).until
        (lambda driber: self.driver.find_element_by_xpath("//*[@id='plist']").is_displayed())

        products_name = self.driver.find_elements_by_xpath("//*[@id='plist']//li/div[2]")
        
        for name in products_name:
            print name.text
            self.assertTrue(name.text.startswith('京东12.12'))
    
    def initiateElement(self):
    #     by id
        self.input_search = self.driver.find_element_by_id('key');
    #     by xpath
        self.button_search = self.driver.find_element_by_xpath("//*[@id='search']//input[@type='button']")

if __name__ == '__main__':
    unittest.main()
#    test = SearchPoductsTest("testProductsNameEqual")
#    runner = unittest.TextTestRunner()
#    runner.run(test)
